.github/workflows/terraform-smoke.yml

```diff
@@ -0,0 +1,90 @@
+name: "Terraform Provider for Juju Smoke"
+on:
+  push:
+    branches: [2.9, 3.1, 3.2, 3.3, 3.4]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+#    paths:
+#      DON'T SET - these are "required" so they need to run on every PR
```

Comment:
Do we actually want this to be a required job?

```
    paths:
      - '**.go'
      - 'go.mod'
      - '.github/workflows/terraform-smoke.yml'
```

Reply:
Eventually I'd like too, but we can put in these for now.